src/Analyser/SpecifiedTypes.php

```diff
@@ -114,7 +114,9 @@ public function unionWith(SpecifiedTypes $other): self
 			];
 		}
 
-		return new self($sureTypeUnion, $sureNotTypeUnion);
```

Comment:
Does it make sense to extract this in a dedicated fix PR that can be merged independently?

Reply:
agree. want to hear from ondrey first, whether the change itself is acceptable